src/bitcoinForks/index.ts

```diff
@@ -745,17 +748,22 @@ export default class BitcoinWallet implements Partial<IWallet> {
 
     const txrefs = response.data || [];
 
-    txrefs.forEach((txref: any) => {
-      const utxo = {
-        address: txref.address,
-        txId: txref.txid,
-        vout: txref.vout,
-        value: parseInt(txref.value, 10),
-        block_height: txref.height,
-        confirmations: txref.confirmations
-      };
-      utxos.push(utxo);
-    });
+    await Promise.all(
+      txrefs.map(async (txref: any) => {
+        const utxo = {
+          address: txref.address,
+          txId: txref.txid,
+          vout: txref.vout,
+          value: parseInt(txref.value, 10),
+          block_height: txref.height,
+          confirmations: txref.confirmations
+        };
+        const transaction = await this.transactionDB?.getOne({
+          hash: txref.txid
+        });
```

Comment:
Throw error when transactionDb is not present. Check how addressDB was mandated on required functions.

Reply:
added the check in all relevant functions